kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala

```diff
@@ -645,6 +647,51 @@ object KyuubiConf {
       .stringConf
       .createWithDefault("uid")
 
+  val AUTHENTICATION_JDBC_DRIVER: OptionalConfigEntry[String] =
+    buildConf("kyuubi.authentication.jdbc.driver.class")
+      .doc("Driver class name for JDBC Authentication Provider.")
+      .version("1.6.0")
+      .stringConf
+      .createOptional
+
+  val AUTHENTICATION_JDBC_URL: OptionalConfigEntry[String] =
+    buildConf("kyuubi.authentication.jdbc.url")
+      .doc("JDBC URL for JDBC Authentication Provider.")
+      .version("1.6.0")
+      .stringConf
+      .createOptional
+
+  val AUTHENTICATION_JDBC_USERNAME: OptionalConfigEntry[String] =
+    buildConf("kyuubi.authentication.jdbc.username")
+      .doc("Database username for JDBC Authentication Provider.")
+      .version("1.6.0")
+      .stringConf
+      .createOptional
+
+  val AUTHENTICATION_JDBC_PASSWORD: OptionalConfigEntry[String] =
+    buildConf("kyuubi.authentication.jdbc.password")
+      .doc("Database password for JDBC Authentication Provider.")
+      .version("1.6.0")
+      .stringConf
+      .createOptional
+
+  val AUTHENTICATION_JDBC_QUERY: OptionalConfigEntry[String] =
+    buildConf("kyuubi.authentication.jdbc.query")
+      .doc("Query SQL template with placeholders " +
+        "for JDBC Authentication Provider to execute. " +
+        "Authentication passes if at least one row fetched in the result set." +
+        "Available placeholders are: <ul>" +
+        "<li>`${username}`</li>" +
+        "<li>`${password}`</li></ul>" +
+        "eg.: query sql `SELECT 1 FROM auth_table WHERE user=${username} AND " +
+        "passwd=MD5(CONCAT(salt,${password}));` " +
+        "will be prepared as: `SELECT 1 FROM auth_table " +
+        "WHERE user=? AND passwd=MD5(CONCAT(salt,?));`" +
+        " with value replacement of `username` and `password` in string type.")
+      .version("1.6.0")
+      .stringConf
+      .createOptional
```

Comment:
Maybe a default SQL is better. cc @yaooqinn @turboFei WDYT?

`SELECT 1 FROM kyuubi_user WHERE user=${username} AND passwd=MD5(${password})`

Reply:
A default sql may be not a proper way to suggest. Explict query sql is the key to make use of db , including the hash or encrypt/decrypt funcion, the table(s) for authentication.

Reply:
I'm neutral here.
But for SQL dialect concerns, md5 is a common function supported by most of dbms, ref: https://github.com/apache/spark/pull/37169. But in other hand, md5 w/o salt is not security

Reply:
yes, got your idea. Thus the compatibility is still limited , eg. Md5 is not supported in Derby. Plus we are not suggesting users to choose in hash or encryptions, but rather than encourage them to maximize usage in conditions, rules, joins and methods in their senarios, eg. limit max 5 times logs in 1 min with specific prefix in username or different types of password or tokens.

Reply:
ok, let's leave it then.

Reply:
> I'm neutral here. For SQL dialect concerns, md5 is a common function supported by most of dbms, ref: [apache/spark#37169](https://github.com/apache/spark/pull/37169). But in other hand, md5 w/o salt is not security

alright. and thx for the link to [apache/spark#37169](https://github.com/apache/spark/pull/37169). it is clear and helpful.